src/formatDate.js

```diff
@@ -7,8 +7,70 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDateArr = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  const separator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateArr = date.split(separator);
+
+  const requiredYearFormat = toFormat.find(
+    (element) => element === 'YYYY' || element === 'YY',
+  );
+
+  const changeYearFormat = () => {
+    if (requiredYearFormat.length === 4 && year.length === 4) {
+      return year;
+    }
+
+    if (requiredYearFormat.length === 4 && year < 30) {
+      return `20${year}`;
+    }
+
+    if (requiredYearFormat.length === 4 && year >= 30) {
+      return `19${year}`;
+    }
+
+    if (requiredYearFormat.length === 2) {
+      return year.slice(-2);
+    }
+  };
```

Comment:
The 'changeYearFormat' function is defined inside the 'formatDate' function, which can lead to the 'changeYearFormat' function being redefined every time 'formatDate' is called. Consider moving 'changeYearFormat' outside of 'formatDate' to avoid unnecessary redefinitions and to keep the 'formatDate' function cleaner.